akka-http-core/src/main/scala/akka/http/impl/engine/client/OutgoingConnectionBlueprint.scala

```diff
@@ -106,126 +108,109 @@ private[http] object OutgoingConnectionBlueprint {
 
   // a simple merge stage that simply forwards its first input and ignores its second input
   // (the terminationBackchannelInput), but applies a special completion handling
-  class TerminationMerge
-    extends FlexiMerge[HttpRequest, FanInShape2[HttpRequest, HttpResponse, HttpRequest]](new FanInShape2("TerminationMerge"), Attributes.name("TerminationMerge")) {
-    import FlexiMerge._
+  private object TerminationMerge extends GraphStage[FanInShape2[HttpRequest, HttpResponse, HttpRequest]] {
+    private val requests = Inlet[HttpRequest]("requests")
+    private val responses = Inlet[HttpResponse]("responses")
+    private val out = Outlet[HttpRequest]("out")
 
-    def createMergeLogic(p: PortT) = new MergeLogic[HttpRequest] {
+    val shape = new FanInShape2(requests, responses, out)
 
-      val requestInput = p.in0
-      val terminationBackchannelInput = p.in1
+    override def createLogic = new GraphStageLogic(shape) {
+      passAlong(requests, out, doFinish = false, doFail = true)
+      setHandler(out, eagerTerminateOutput)
```

Comment:
`out` is the output, which just gets the elements from the `requests` input; the `responses` input is indeed continuously pulled a few lines below this one